lib/charms/vault_k8s/v0/vault_managers.py

```diff
@@ -816,3 +829,243 @@ def get_provider_vault_token(
                     label=self.AUTOUNSEAL_TOKEN_SECRET_LABEL,
                 )
         return external_vault.token
+
+
+class PKIManager:
+    """Encapsulates the business logic for managing PKI certificates in Vault from a Charm."""
+
+    def __init__(
+        self,
+        charm: CharmBase,
+        vault_client: VaultClient,
+        common_name: str,
+        mount_point: str,
+        role_name: str,
+        vault_pki: TLSCertificatesProvidesV4,
+        tls_certificates_pki: TLSCertificatesRequiresV4,
+    ):
+        """Create a new PKIManager object.
+
+        Args:
+            charm: The charm this manager is associated with
+            vault_client: The Vault client object
+            common_name: The common name for the certificates allowed by the PKI backend
+            mount_point: The mount point in Vault for the PKI backend
+            role_name: The role name for the PKI backend
+            vault_pki: The vault_pki provider relation helper library
+            tls_certificates_pki: The tls_certificates_pki requirer relation helper library
+        """
+        self.validate_input(common_name=common_name)
+        self._vault_client = vault_client
+        self._juju_facade = JujuFacade(charm)
+        self._common_name = common_name
+        self._mount_point = mount_point
+        self._role_name = role_name
+        self._vault_pki = vault_pki
+        self._tls_certificates_pki = tls_certificates_pki
+
+    def intermediate_ca_cert_request(self) -> CertificateRequestAttributes:
+        """Return the certificate request attributes for the intermediate CA.
+
+        Returns:
+            The CertificateRequestAttributes that should be used to request the
+            intermediate CA certificate from the TLS certificates provider.
+        """
+        return CertificateRequestAttributes(
+            self._common_name,
+            is_ca=True,
+        )
+
+    def validate_input(self, *, common_name: str):
+        """Validate the inputs to this class."""
+        if common_name == "":
+            raise ManagerError("Common name cannot be empty.")
```

Comment:
I'm indifferent about this, but I wanted it to error if it was empty. It actually saved me once already.

An assert or just a simple check would work, but I thought I would start something extensible in case there's other things we wanted to validate.

Reply:
This becomes irrelevant if we do https://github.com/canonical/vault-k8s-operator/pull/559#discussion_r1890760925